packages/react-router/src/link.tsx

```diff
@@ -258,7 +258,7 @@ export type ResolveRoute<
   TTo,
   TPath = ResolveRelativePath<TFrom, TTo>,
 > = TPath extends string
-  ? string extends TTo
+  ? TFrom extends TPath
```

Comment:
If the resolved path is the same as `TFrom` then we know we should use `RouteByPath`